extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/MicroProfileRestClientRequestFilter.java

```diff
@@ -72,6 +72,7 @@ public void filter(ClientRequestContext requestContext) {
                 reactiveRequestContext.suspend();
                 MultivaluedMap<String, String> outgoingHeaders = incomingHeaders;
                 reactiveClientHeadersFactory.getHeaders(incomingHeaders).subscribe().with(newHeaders -> {
+                    outgoingHeaders.clear();
```

Comment:
Can you explain why this is needed? Prefereably in a source-code comment so it's persistent :)

Reply:
that's the way the factory in the MP spec works. It takes in headers that come from an incoming request and creates a new map of headers from it. It's basically about propagating selected headers.
I'll add a comment about it. But I started to have some doubts re how this code works with calls that do blocking stuff so will first do some experiments :)